ui/docs/reference/integrations/github/checks.mdx

```diff
@@ -0,0 +1,43 @@
+# Taskcluster-Github Checks
+
+Taskcluster-Github supports [Github's Checks API](https://developer.github.com/v3/checks/).
+At the moment, the feature is in beta. If you would like to try it out:
+* Make sure you use `.taskcluster.yml` v1
+* Add `reporting: checks-v1` line at the root level of your `.taskcluster.yml`
+* You will need `queue:route:checks` scope
+* If you are using a decision task, make sure you add `route.checks` to the `routes` of your tasks
+
+## Custom Output in Checks
+
+If you need to add some sort of custom output to your checks (for example, you would like to process
+your logs and add some results to the check's body):
+* You need to arrange for an artifact with the custom text (plain or Markdown) you'd like to add to your check 
+ (for the above example, you arrange the processing of your logs in the way you want, 
+ format the result the way you want using Markdown and save the result in an artifact)
+* Add something like 
+    ```YML
+    extra:
+        github:
+            customCheckRun:
+                textArtifactName: <artifact_name>
+    ```
+ to the definition of the task for which you want customize the check.
+* If it's OK with you that the artifact is public, make sure to start the name of your artifact with
+ `public/`, like this: `public/<file_name>` both when you create the artifact and reference it in your
+ task definition. In this case, you will not need any scopes.
+* If you don't want the artifact to be public, name your artifact simply `<file_name>` both when creating it
+ and referencing it in your task definition. In this case, you will need a `queue:get-artifact:<file_name>`
+ scope. Add it to the `scopes` section of the same task.
```

Comment:
It doesn't look like this is supported as currently written.

Reply:
Can you explain?

Reply:
There is no support for private artifacts.

Reply:
Oh. That was in the `getArtifact` and `createArtifact` documentation. (from the queue code it seemed to me the artifact would be saved in the private bucket for the account (I was looking [here](https://github.com/taskcluster/taskcluster/blob/master/services/queue/src/artifacts.js#L176) ) So, what happens in practice if your artifact name doesn't have `public/` prefix?

Reply:
As the PR is written, tc-github will try to fetch the artifact and fail because it does not use authentication to fetch it.

Note that I'm not saying this must support private artifacts -- I'd be just as happy to see the documentation say that the artifact must be a public artifact.

Reply:
Do you mean [this auth here](https://github.com/taskcluster/taskcluster/pull/2477/files#diff-41e7ecfbb590e5f3bbddd300fbf410dbR499-R505) will not work?

Reply:
Right.

Reply:
>  There's the potential here to leak private artifacts, so it will need some careful thought.

Do you have suggestions on how to improve the current auth?

Reply:
`buildSignedURL` will build a signed URL that can be used for private artifacts.

Reply:
It is not clear to me that that the scopes of the task are the right thing to consult, when checking to see if the artifact should be used. The main reason is that existing tasks that have scopes for private artifacts shouldn't implicitly allow tc-github to use those artifacts for check.

My inclination would be to not support private artifacts in the initial implementation.

Reply:
I asked Fenix folks what their needs are. as far as I am aware _at the moment_, not supporting private artifacts in the initial implementation should be fine

Reply:
update: yeah, they confirmed they do not see the use for private artifacts at the moment. Ok, I will go ahead and follow @tomprince 's advice then because it makes perfect sense to me. We can discuss security of this in more detail down the road and I can always add this back in.